src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -477,6 +478,7 @@ int goto_analyzer_parse_optionst::doit()
 /// Depending on the command line mode, run one of the analysis tasks
 int goto_analyzer_parse_optionst::perform_analysis(const optionst &options)
 {
+  adjust_float_expressions(goto_model);
```

Comment:
I'm honestly not 100% sure how goto-analyzer works, but 

```
#include <assert.h>
#include <fenv.h>

float div_by_ten(float x) {
  return x/10.0f;
}

int main(void)
{
  float f = 1.0f/10.0f;
  __CPROVER_rounding_mode = 1;
  float g = 1.0f/10.0f;
  assert(g < f);
  __CPROVER_rounding_mode = 3;
  float h = 1.0f/10.0f;
  assert(g == h);
  assert(h < f);
  __CPROVER_rounding_mode = 2;
  float i = 1.0f/10.0f;
  assert(f == i);
}
```

putting this into `goto-analyzer --simplify` followed by `--dump-c` gives me

```
Reading GOTO program from `test.out'
#include <assert.h>

// div_by_ten
// file test.c line 4
float div_by_ten(float x);

// div_by_ten
// file test.c line 4
float div_by_ten(float x)
{
  return FLOAT/(x, 10.0f, __CPROVER_rounding_mode);
}

// main
// file test.c line 8
signed int main(void)
{
  float f=0.1f;
  __CPROVER_rounding_mode = 1;
  float g=0.1f;
  /* assertion g < f */
  assert((_Bool)1);
  __CPROVER_rounding_mode = 3;
  float h=0.1f;
  /* assertion g == h */
  assert((_Bool)1);
  /* assertion h < f */
  assert((_Bool)1);
  __CPROVER_rounding_mode = 2;
  float i=0.1f;
  /* assertion f == i */
  assert((_Bool)1);
}
```

Reply:
The problem is that it has the (internal) rounded float exprs rather than the correct ones.  Please adjust the float expressions during evaluation, not in the program.

Reply:
Maybe @martin-cs will hate me for that, but ... how about a final cleanup step that removes all the internal ones? Much like is now done for `remove_returns` -> `restore_returns`.

Reply:
The expression is already being modified during evaluation (substituting constants, calling simplify, etc.), so one more modification there won't hurt.  Doing it up front requires changing it one way and then changing it back at the end.

Reply:
Ok, but I have to clearly admit that it was me pushing @hannes-steffenhagen-diffblue in this direction with the rationale that repeated modifications would seem costly. With the added learnings ever since I am happy to backtrack on this. My bad.

Reply:
Also my apologies to @hannes-steffenhagen-diffblue for not being more present in this one.